### PR TITLE
Fix nlp_example

### PR DESCRIPTION
**Pull request by jjiang2cal**
loss.backward does gradient all_reduce in pytorch DDP. In gradient
accumulation, we do not want all_reduce in every step, so use
no_sync in DDP to avoid unnecessary gradient sync.

### CODE REVIEW

**Comment by sgugger** [comment]
That's completely true, but the goal of the example is to be as simple as possible and it does work with computing the gradients at each step, so I'm a bit torn on whether we should have this or not.

What do you think @LysandreJik @thomwolf ?

**Comment by lvwerra** [comment]
If not in the example it would be great if there would a hint/warning in the documentation somewhere. We made that mistake in one of our training script because of just copy-pasting the example.

**Comment by sgugger** [comment]
Accelerate will soon handle gradient accumulation directly, specifically for this problem :-)
